src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1987,13 +1977,28 @@ private object GetMessageJson(JsonRpcMessage message)
             }
         }
 
-#pragma warning disable AvoidAsyncSuffix // Avoid Async suffix
-        private ValueTask TransmitAsync(JsonRpcMessage message, CancellationToken cancellationToken)
-#pragma warning restore AvoidAsyncSuffix // Avoid Async suffix
+        private async ValueTask TransmitAsync(JsonRpcMessage message, CancellationToken cancellationToken)
         {
-            this.TraceMessageSent(message);
+            try
+            {
+                this.TraceMessageSent(message);
+                await this.MessageHandler.WriteAsync(message, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                if ((bool)(this.MessageHandler as IDisposableObservable)?.IsDisposed)
+                {
+                    var e = new JsonRpcDisconnectedEventArgs(
+                        string.Format(CultureInfo.CurrentCulture, Resources.ErrorWritingJsonRpcResult, exception.GetType().Name, exception.Message),
```

Comment:
You're using `Resources.ErrorWritingJsonRpcResult` which is specific to result messages, but this could be an error in writing *any* message. Can you define a new string called `ErrorWritingJsonRpcMessage` that is slightly more generic to use here?